library/core/src/intrinsics.rs

```diff
@@ -815,6 +815,7 @@ extern "rust-intrinsic" {
     /// This will statically either panic, or do nothing.
     ///
     /// This intrinsic does not have a stable counterpart.
+    #[rustc_const_unstable(feature = "const_maybe_assume_init", issue = "none")]
```

Comment:
you implemented the two intrinsics below in the interpreter, but they have no const uses yet. Is there a reasonable method using them that we could unstably make `const fn`, too?

Reply:
From a grep i found that [`mem::zeroed`](https://github.com/rust-lang/rust/blob/master/library/core/src/mem/mod.rs#L620) and [`mem::uninitialized`](https://github.com/rust-lang/rust/blob/master/library/core/src/mem/mod.rs#L656) are using their respective intrinsic. I believe `mem::uninitialized` would be possible to make const thanks to a `MaybeUninit::assume_init`. However `mem::zeroed` depends on [`MaybeUninit::zeroed`](https://github.com/rust-lang/rust/blob/master/library/core/src/mem/maybe_uninit.rs#L361).

Does it make sense to make `mem::uninitialized` `const` since it is marked as deprecated?

Reply:
> Does it make sense to make `mem::uninitialized` `const` since it is marked as deprecated?

No, you're absolutely right.

So, I think we should not implement the other intrinsics in the interpreter then. We can implement them once there's a real use case for them.